### PR TITLE
Add Zebra System simulation with AND|OR block support

### PR DESCRIPTION
**Pull request by DiamondJoseph**
- Update Zebra implementation for newer adapter/system simulation behaviours
- Add tests for blocks and dummy config
- Zebra support with 4 each AND/OR blocks, each with 4 inputs
- Supports getting/setting params/muxes from adapter

### CODE REVIEW

File: src/tickit_devices/zebra/zebra.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    mux_types,
+    param_types,
+    register_names,
+)
+
+
+class ZebraAdapter(BaseSystemSimulationAdapter):
+    _components: Dict[ComponentID, DeviceSimulation]
+    params: dict[str, int]
+    """network adapter for zebra system simulation"""
```
Comment (abbiemery) [inline src/tickit_devices/zebra/zebra.py:22]: Perhaps describe a bit more on what it does and how to make it do it.

**Review by abbiemery — APPROVED** [review]
Looks like a good start to a zebra. I can't see anything obviously wrong, but I don't know how Zebras work so perhaps not the best person to judge. It works and passes the tests, so I'm happy enough for it to be added and we can develop it from there, fixing any issues that might crop up along the way.

File: src/tickit_devices/zebra/__init__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from tickit.core.typedefs import ComponentID, ComponentPort, PortID
+
+from tickit_devices.zebra._common import default_filler, param_types
+from tickit_devices.zebra.and_or_block import AndOrBlockConfig
+from tickit_devices.zebra.zebra import ZebraAdapter
+
+
+@pydantic.v1.dataclasses.dataclass
+class Zebra(ComponentConfig):
+    """Zebra simulation with TCP server."""
```
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply src/tickit_devices/zebra/__init__.py]: Added verbosity

File: src/tickit_devices/zebra/zebra.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    def __init__(self, params: dict[str, int]):
+        self.params = params
+
+    def setup_adapter(
+        self,
+        components: Dict[ComponentID, DeviceSimulation],
+        wiring: Union[Wiring, InverseWiring],
+    ) -> None:
+        """Provides the components and wiring of a SystemSimulationComponent."""
```
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply src/tickit_devices/zebra/zebra.py]: Added some docs for the changed behaviour vs. the super

File: src/tickit_devices/zebra/zebra.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        if reg_name in self.params:
+            self.params[reg_name] = value_int
+
+            for block_name in param_types[reg_name].blocks:
+                await asyncio.create_task(
+                    self._components[block_name].raise_interrupt()
+                )
+
+        else:
+            # TODO: wire adapter mux to block inputs
```
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply src/tickit_devices/zebra/zebra.py]: Have left muxes to be done when rewiring devices and circular paths are allowed

**Review by abbiemery — APPROVED** [review]
Looks good as our starting point.  I think It would be worth merging it as is then we can discus with Dom and Tom about use/developments from here and iterate from this.

**Pull request merged**